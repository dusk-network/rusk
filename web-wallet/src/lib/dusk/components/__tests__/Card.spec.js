import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { Card } from "..";

describe("Card", () => {
  afterEach(cleanup);

  it("renders the Card component with a heading", () => {
    const { container } = render(Card, { props: { heading: "My Card" } });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the Card component with an icon when iconPath is provided", () => {
    const { container } = render(Card, {
      props: { heading: "My Card", iconPath: "M3,3H21V21H3V3M5,5V19H19V5H5Z" },
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
