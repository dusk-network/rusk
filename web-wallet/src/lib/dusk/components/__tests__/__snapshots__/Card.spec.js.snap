// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders the Card component with a heading 1`] = `
<div>
  <div
    class="dusk-card"
  >
    <header
      class="dusk-card__header"
    >
       
      <h3
        class="h4"
      >
        My Card
      </h3>
       
    </header>
     
  </div>
  
  <!--&lt;Card&gt;-->
</div>
`;

exports[`Card > renders the Card component with an icon when iconPath is provided 1`] = `
<div>
  <div
    class="dusk-card"
  >
    <header
      class="dusk-card__header"
    >
      <svg
        class="dusk-icon dusk-icon--size--normal dusk-card__icon"
        role="graphics-symbol"
        viewBox="0 0 24 24"
      >
        <path
          d="M3,3H21V21H3V3M5,5V19H19V5H5Z"
        />
      </svg>
      
      <!--&lt;Icon&gt;-->
       
      <h3
        class="h4"
      >
        My Card
      </h3>
       
    </header>
     
  </div>
  
  <!--&lt;Card&gt;-->
</div>
`;
